vent/coordinator/coordinator.py

```diff
@@ -72,127 +62,146 @@ def __init__(self, sim_mode=False):
+        return self.control_module.get_control(control_setting_name)
 
     def start(self):
-        # TODO: why add this? Local model doesn't need IPC
-        # if not self.control_module.running():
-        #     self.do_process(IPCMessageCommand.START)
-        while not self.stopping.is_set():
-            sensor_values = self.control_module.get_sensors()
-            self.lock.acquire()
-            self.COPY_sensor_values = sensor_values
-            self.last_message_timestamp = sensor_values.timestamp
-            self.lock.release()
-            for name in [ValueName.PIP,
-                         ValueName.PIP_TIME,
-                         ValueName.PEEP,
-                         ValueName.BREATHS_PER_MINUTE,
-                         ValueName.INSPIRATION_TIME_SEC]:
-                self.lock.acquire()
-                in_control_settings = name not in self.COPY_control_settings
-                self.lock.release()
-                if in_control_settings:
-                    control_setting = self.control_module.get_control(name)
-                    self.lock.acquire()
-                    self.COPY_control_settings[name] = control_setting
-                    self.lock.release()
-
-                self.lock.acquire()
-                disagreed_tentative = name in self.COPY_tentative_control_settings and self.COPY_tentative_control_settings[name] != self.COPY_control_settings[
-                            name]
-                self.lock.release()
-                if disagreed_tentative:
-                    self.lock.acquire()
-                    tentative_control_setting = self.COPY_tentative_control_settings[name]
-                    self.lock.release()
-                    self.control_module.set_control(tentative_control_setting)
-                    self.lock.acquire()
-                    self.COPY_control_settings[name] = self.control_module.get_control(name)
-                    self.lock.release()
-            # sleep 10 ms
-            time.sleep(0.01)
+        """
+        Start the coordinator.
+        This does a soft start (not allocating a process).
+        """
+        self.control_module.start()
+
+    def is_running(self) -> bool:
+        """
+        Test whether the whole system is running
+        """
+        return self.control_module._running
 
     def stop(self):
-        self.stopping.set()
+        """
+        Stop the coordinator.
+        This does a soft stop (not kill a process)
+        """
+        self.control_module.stop()
 
 
 class CoordinatorRemote(CoordinatorBase):
     def __init__(self, sim_mode=False):
         super().__init__(sim_mode=sim_mode)
-        # TODO: pass max_heartbeat_interval
-        self.process_manager = ProcessManager()
-        self.rpc = IPC()
+        # TODO: according to documentation, pass max_heartbeat_interval?
+        self.process_manager = ProcessManager(sim_mode)
+        self.proxy = get_rpc_client()
```

Comment:
"proxy" variable name doesn't really capture what this is. I'd prefer something like "rpc_client".

Reply:
Good idea!